Source/WebCore/svg/SVGLocatable.cpp

```diff
@@ -25,29 +25,37 @@
 
 #include "RenderElement.h"
 #include "SVGElementTypeHelpers.h"
+#include "SVGForeignObjectElement.h"
 #include "SVGGraphicsElement.h"
 #include "SVGImageElement.h"
 #include "SVGLayerTransformComputation.h"
 #include "SVGMatrix.h"
 #include "SVGNames.h"
+#include "SVGSVGElement.h"
+#include "SVGSymbolElement.h"
 #include "TransformState.h"
 
 namespace WebCore {
 
-static bool isViewportElement(Node* node)
+static SVGElement* toViewportElement(Element& element)
 {
-    return (node->hasTagName(SVGNames::svgTag)
-        || node->hasTagName(SVGNames::symbolTag)
-        || node->hasTagName(SVGNames::foreignObjectTag)
-        || is<SVGImageElement>(*node));
+    if (auto* svgElement = dynamicDowncast<SVGSVGElement>(element))
+        return svgElement;
+    if (auto* svgElement = dynamicDowncast<SVGSymbolElement>(element))
+        return svgElement;
+    if (auto* svgElement = dynamicDowncast<SVGForeignObjectElement>(element))
+        return svgElement;
+    if (auto* svgElement = dynamicDowncast<SVGImageElement>(element))
+        return svgElement;
+    return nullptr;
 }
 
 SVGElement* SVGLocatable::nearestViewportElement(const SVGElement* element)
```

Comment:
`SVGElement::viewportElement()` is very similar to this function except it does not check for foreign object. I would suggest leaving this method as is for now and address the clarity and the code redundancy in a separate PR.